src/bloqade/builder/field.py

```diff
@@ -64,99 +130,292 @@ def location(
-        """Address a single atom (or multiple) atoms.
+        """Address a a subset of atoms as part of defining the spatial modulation component of a drive
+        with a unique scaling factor for each of them.
+
+        Args:
+            labels (Union[List[int], int]): A list of site indices or a single site index to target.
+            scales (Union[List[ScalarType], ScalarType, None]): A list of scaling factors for each site or a single scaling factor to target all sites.
+
+        Returns:
+            Location: A program node representing the spatial modulation with unique scaling factors for some atoms.
+
+        ??? abstract "Background and Context"
+
+            Fields give you local (single-atom) control over the many-body Rydberg Hamiltonian.
+
+            They are a sum of one or more spatial modulations multiplied by a waveform:
+
+            $$
+            \\begin{align*}
+            & F_{i}(t) = \sum_{\\alpha} C_{i}^{\\alpha}f_{\\alpha}(t) 
+            \\\\
+            & C_{i}^{\\alpha} \in \mathbb{R} 
+            \\\\
+            & f_{\\alpha}(t) \colon \mathbb{R} \\to \mathbb{R}
+            \\end{align*}
+            $$
+
+            The $i$-th component of the field is used to generate the drive at the $i$-th site.
+
+            Note that the drive is only applied if the $i$-th site is filled with an atom.
+
+            You build fields in Bloqade by first specifying the spatial modulation followed by the waveform
+            it should be multiplied by.
+
+            In the case of a spatial modulation produced by `location`, you can provide a unique scaling factor scaling factor $C_{i}^{\\alpha}$ for each OR some sites $i$ in the system.
+
+        ??? example "Examples"
+
+            We start by creating a simple program with 3 atoms targeting the Rydberg level coupling and
+            we begin constructing the Rabi amplitude field:
+            
+            ```python
+            from bloqade import start
+            program = start.add_position([(0,0), (4,0), (8,0)]).rydberg.rabi.amplitude
+            ```
+
+            We can specify that we only want to target the first site with a scaling of $1.0$:
+            ```python
+            single_location_scaled = program.location(0)
+            ```
+
+            We may also specify that we want to target only the first site but with a different scaling:
+            ```python
+            custom_scaled_single_location = program.location(0, 0.5)
+            ```
+
+            We can also target a subset of the sites with a scaling of $1.0$:
+            ```python
+            multi_location_scaled = program.location([0, 2])
+            ```
+
+            And finally we can target a subset of sites with custom scaling coefficients:
+            ```python
+            custom_multi_location_scaled = program.location([0, 2], [0.5, 0.25])
+            ```
+        
+        ??? warning "Potential Pitfalls"
+
+            # Global vs. Local Detuning
```

Comment:
this might be too large after render? <h1> seems to be a pretty large thing to me.

Reply:
I've tested this locally and the rendering seems to be fine on the website. Inside admonitions the text in general seems to be smaller which makes this a reasonable looking headline. 

Check out the screenshot below:
<img width="530" alt="Screenshot 2024-04-29 at 8 21 19 AM" src="https://github.com/QuEraComputing/bloqade-python/assets/32608115/b501517f-a757-4d76-bd09-631916c0c643">

Reply:
this is too large in vscode

<img width="1115" alt="image" src="https://github.com/QuEraComputing/bloqade-python/assets/8445510/3c0a96dc-88fe-453c-b665-d543d1c622ec">

Reply:
I see, you're talking about the headers that are outside the admonition box. I agree it's cartoonishly large in VSCode but I just worry the titles are now too small on MkDocs.

If I use `##` or `###` instead of `#` the MkDocs stuff comes out as the following:
<img width="135" alt="Screenshot 2024-05-01 at 9 09 07 AM" src="https://github.com/QuEraComputing/bloqade-python/assets/32608115/b96df7b8-a9b3-4039-bb47-b23e757561a2">

Instead of: 
<img width="157" alt="Screenshot 2024-05-01 at 9 09 27 AM" src="https://github.com/QuEraComputing/bloqade-python/assets/32608115/bf7bb2fd-415c-42d9-a5c6-7c7d43024298">

Reply:
hmm, there seems to be a conflict here, but IMO vscode is doing the right convention of title size...